dask/dataframe/backends.py

```diff
@@ -333,6 +340,31 @@ def _nonempty_index(idx):
         return pd.RangeIndex(2, name=idx.name, dtype=idx.dtype)
     elif is_any_real_numeric_dtype(idx):
         return typ([1, 2], name=idx.name, dtype=idx.dtype)
+    elif is_arrow_dtype(idx.dtype) and pa.types.is_timestamp(idx.dtype.pyarrow_dtype):
```

Comment:
I'm a little surprised we need to have the `is_arrow_dtype` check here. Naively I was expecting something like 

```python
        return pd.Index(
            [pd.Timestamp("1970-01-01"), pd.Timestamp("1970-01-02")],
            dtype=idx.dtype,
            name=idx.name,
        )
```

To work for both `pyarrow`-backed and non-`pyarrow`-backed dtypes. With `pandas` handling the conversion internally. 

Why not use `pd.api.types.is_*_dtype(...)` checks instead? Maybe there isn't a corresponding check in `pd.api.types` for these dtypes?

Reply:
Also, previously I added something similar to our non-empty meta Series creation that resulted in meta creation working, but with all `pd.NA` values

https://github.com/dask/dask/blob/35554159781b2f5b7fecbd967ea76e90f97cb63c/dask/dataframe/_dtypes.py#L24-L26

It might be worth revisiting / reusing some of the logic here for better series creation. Not saying that needs to happen in this PR. Just mentioned for context and as a possible follow-up

Reply:
I will look into the existing logic tomorrow.

Non-arrow DatetimeIndexes have a freq attribute which would require special casing too. I like the current version a bit better because it's explicit, but can change if you prefer.

Most of the non numeric is_*_dtype aren't arrow aware yet, ``is_datetime64_dtype`` for example isn't. They have quite different semantics, which makes supporting arrow there non-trivial. We ran into a couple of problems over the last few weeks there, not sure if/when we support arrow there. The whole date time-conversion logic in pandas makes this non-trivial.